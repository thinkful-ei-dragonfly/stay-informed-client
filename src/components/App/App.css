body {
  min-width: 320px;
  min-height: 100vh;
}

.container {
  margin-top: 20px;
  justify-content: left; 
  width: 80%;
  margin-left: 10%;
  border-radius: 20px;
  color: #02111B;
  padding: 20px;
}

@media screen and (min-width: 450px) {
  .container {
    width: 50%;
    margin-left: 25%;
  }
}