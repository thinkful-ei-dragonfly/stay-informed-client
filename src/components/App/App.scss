.App {
  padding: 0 25px;
  @media (min-width: 768px) {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
  }

}
.main-app {
  margin-top: 1.5em;
  @media (min-width: 959px) {
    padding: 10px 0;
    margin-top: 0;
  }
}

.wrapper {
  max-width: 1600px;
  margin: 0 auto;
}
