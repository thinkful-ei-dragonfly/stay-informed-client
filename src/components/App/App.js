import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import Header from '../Header/Header'
import PrivateRoute from '../PrivateRoute/PrivateRoute'
import PublicOnlyRoute from '../PublicOnlyRoute/PublicOnlyRoute'
import RegistrationRoute from '../../routes/RegistrationRoute/RegistrationRoute'
import LoginRoute from '../../routes/LoginRoute/LoginRoute'
import DashboardRoute from '../../routes/DashboardRoute/DashboardRoute'
import LearningRoute from '../../routes/LearningRoute/LearningRoute'
import NotFoundRoute from '../../routes/NotFoundRoute/NotFoundRoute'
import './App.css'

export default class App extends Component {
  state = { hasError: false }

  static getDerivedStateFromError(error) {
    console.error(error)
    return { hasError: true }
  }

  render() {
    const { hasError } = this.state
    return (
      <div className='App'>
        <Header />
        <main>
          {hasError && (
            <p>There was an error! Oh no!</p>
          )}
          <Switch>
            <PrivateRoute
              exact
              path={'/'}
              component={DashboardRoute}
            />
            <PrivateRoute
              path={'/learn'}
              component={LearningRoute}
            />
            <PublicOnlyRoute
              path={'/register'}
              component={RegistrationRoute}
            />
            <PublicOnlyRoute
              path={'/login'}
              component={LoginRoute}
            />
            <Route
              component={NotFoundRoute}
            />
          </Switch>
        </main>
      </div>
    );
  }
}
