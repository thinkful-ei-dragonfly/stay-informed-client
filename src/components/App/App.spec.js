import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'

describe(`App component`, () => {
  it('renders a form.App by default', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
