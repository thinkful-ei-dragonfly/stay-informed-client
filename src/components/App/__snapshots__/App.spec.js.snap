// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders a form.App by default 1`] = `
<div
  className="App"
>
  <Header />
  <div
    className="main-app"
  >
    <Switch>
      <PublicOnlyRoute
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/dashboard"
      />
      <AllOtherRoute
        component={[Function]}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/dashboard"
      />
      <Route
        component={[Function]}
        path="/voter-registration"
      />
      <Route
        component={[Function]}
        path="/representatives/:repId"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/register"
      />
      <PublicOnlyRoute
        component={[Function]}
        path="/login"
      />
      <PrivateRoute
        component={[Function]}
        path="/user/:userId"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
  <Footer />
</div>
`;
